File: app/src/main/java/com/samsalek/foodbankandroid/model/Dish.java
package com.samsalek.foodbankandroid.model;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.samsalek.foodbankandroid.jackson.LocalDateTimeDeserializer;
import com.samsalek.foodbankandroid.jackson.LocalDateTimeSerializer;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Dish {

    private String id;

    private String name;
    private String category = "Unknown";
    private List<String> ingredients = new ArrayList<>();
    private List<String> instructions = new ArrayList<>();

    @JsonSerialize(using = LocalDateTimeSerializer.class)
    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    private LocalDateTime creationTime = LocalDateTime.now();
}

File: app/src/main/java/io/github/samsalmag/foodbankandroid/model/Dish.kt
package io.github.samsalmag.foodbankandroid.model

import com.fasterxml.jackson.databind.annotation.JsonSerialize

import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import io.github.samsalmag.foodbankandroid.jackson.LocalDateTimeDeserializer
import io.github.samsalmag.foodbankandroid.jackson.LocalDateTimeSerializer

import java.time.LocalDateTime

data class Dish (
    val id: String,
    var name: String,
    var categories: MutableList<String>,
    val ingredients: MutableList<Ingredient>,
    val instructions: String,

    @JsonSerialize(using = LocalDateTimeSerializer::class)
    @JsonDeserialize(using = LocalDateTimeDeserializer::class)
    val creationTime: LocalDateTime,

    @JsonSerialize(using = LocalDateTimeSerializer::class)
    @JsonDeserialize(using = LocalDateTimeDeserializer::class)
    var latestModTime: LocalDateTime
) {
    constructor(): this("", "", mutableListOf(), mutableListOf(), "", LocalDateTime.now(), LocalDateTime.now())
}
